### ISSUE TITLE
softmax error

### ISSUE THREAD

**Issue by 773041642** [NVIDIA/TensorRT#2021]
![image](https://user-images.githubusercontent.com/56188697/171450851-2b1bf25e-1bc5-4245-b163-ff0629868681.png)

**TensorRT Version**:TensorRT 8.4EA 
**NVIDIA GPU**: NVIDIA 2080T
**CUDA Version**: cuda10.2
**Operating System**: windows  

I made this error when I increased the input tensor size to (1,65536,6). Does Softmax need larger memory to run, or is it caused by some fusion operations

**Comment by nvpohanh** [comment]
@773041642 It looks like a TRT bug. Could you share the ONNX file you used or the code you use? Thanks

**Comment by 773041642** [comment]
> @773041642 It looks like a TRT bug. Could you share the ONNX file you used or the code you use? Thanks

I repeated the error with the following code:
![image](https://user-images.githubusercontent.com/56188697/171544060-e2e14e45-6935-4d4f-b22b-b6056da8ec2c.png)

Error is as follows:
![image](https://user-images.githubusercontent.com/56188697/171544191-7d0356fc-2ab2-4da6-8e07-d299bdc6f390.png)

**Comment by nvpohanh** [comment]
Thanks for the code. We will debug this internally.

Meanwhile, could you do an experiment for me? If you set the input size to `{1, 65535, 16, 16}`, does it still fail?

**Comment by 773041642** [comment]
> Thanks for the code. We will debug this internally.
> 
> Meanwhile, could you do an experiment for me? If you set the input size to `{1, 65535, 16, 16}`, does it still fail?

when i set the input size to `{1, 65535, 16, 16}` , it  can work.

 I did a couple of other experiments, For an input of size `{a, b, 16 ,16}`, if a*b ≥ 65536, it can't work.

**Comment by nvpohanh** [comment]
@773041642 Thanks. I can repro the issue internally and we will debug/fix this.

Btw, which axis do you want to do softmax on? You are using `0x4` which means you are doing softmax on the first 16 of the `{1, 65536, 16 (<-this axis), 16}`. Is this what you want? If you want to do softmax over the "65536" dimension, the axes should be set to `1 << 1` which is `0x02`.

**Comment by 773041642** [comment]
> @773041642 Thanks. I can repro the issue internally and we will debug/fix this.
> 
> Btw, which axis do you want to do softmax on? You are using `0x4` which means you are doing softmax on the first 16 of the `{1, 65536, 16 (<-this axis), 16}`. Is this what you want? If you want to do softmax over the "65536" dimension, the axes should be set to `1 << 1` which is `0x02`.

Thanks for your reply. When I set the axes to be 0x02, engine can be built.

But the code in PyTorch looks like this `nn.Softmax(dim=-2)`, so I must do softmax on the first 16.

**Comment by nvpohanh** [comment]
I see. That makes sense.

I have created an internal tracker (id: 3668379). Will let you know when we fix the issue. Thanks for reporting this.

**Comment by nvpohanh** [comment]
This has been fixed and will be available in TRT 8.5 release. Thanks for reporting this issue!
